File: frontend/src/types/media.ts
export interface MediaFile {
  id: string
  project_id: string
  filename: string
  file_path: string
  media_type: 'video' | 'audio' | 'image'
  duration_ms: number | null
  width: number | null
  height: number | null
  fps: number | null
  codec: string | null
  file_size: number | null
  has_audio: boolean
  is_deleted: boolean
  created_at: string
}

export interface MergeRequest {
  video_media_id: string
  audio_media_id: string
  output_name?: string
}

File: backend/app/schemas/media.py
from datetime import datetime

from pydantic import BaseModel


class MediaResponse(BaseModel):
    id: str
    project_id: str
    filename: str
    file_path: str
    media_type: str
    duration_ms: int | None = None
    width: int | None = None
    height: int | None = None
    fps: float | None = None
    codec: str | None = None
    file_size: int | None = None
    has_audio: bool = False
    is_deleted: bool = False
    created_at: datetime

    model_config = {"from_attributes": True}


class MergeRequest(BaseModel):
    video_media_id: str
    audio_media_id: str
    output_name: str = "merged_output.mp4"
